docs/guides/parallel-scraping/parallel-scraping.mdx

```diff
@@ -0,0 +1,220 @@
+---
+id: parallel-scraping-guide
+title: Parallel Scraping Guide
+sidebar_label: Parallel Scraping
+description: Parallelizing your scrapers with Crawlee
+---
+
+import ApiLink from '@site/src/components/ApiLink';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+import SharedSource from '!!raw-loader!./shared.mjs';
+import AdaptedRoutesSource from '!!raw-loader!./adapted-routes.mjs';
+import ParallelScraperSource from '!!raw-loader!./parallel-scraper.mjs';
+import ModifiedDetailRouteSource from '!!raw-loader!./modified-detail-route.mjs';
+
+:::warning Experimental features ahead
+
+At the time of writing this guide (December 2023), request locking is still an experimental feature. You can read more about the experiment by visiting the [request locking experiment](../experiments/experiments-request-locking) page.
+
+:::
+
+In this guide, we will walk through how you can turn your single scraper into a scraper that can be parallelized and run in multiple instances. This guide
```

Comment:
It might be a personal preference, but I like to understand the problem before diving into a full article on a how-to. This helps determine if the subject might be related to what I'm looking for, especially when skimming through the docs.

In that sense, I suggest starting by introducing the problem, something like:
> Node.js is single-threaded for asynchronous I/O operations, but CPU-intensive tasks (like scraping with crawlee) can block the event loop, leading to performance issues. In such cases, running multiple Node.js scripts (or processes) can help. This approach aligns with the concept of horizontal scaling.

Then stating the "drawbacks" of the solution:

> However, there are considerations:

> - Complexity: Managing multiple processes adds complexity in terms of deployment, process management, inter-process communication, and error handling. For production workloads, you might want to use more robust tools like PM2 or a container orchestrator such as Kubernetes.
> - Resource Utilization: More processes mean more usage of system resources. You need to ensure that your system has enough memory and CPU resources.
> - Consistency and State Management: Since crawlers are running independently in their own process, they need to communicate to distribute the workload (the crawling requests) and thus require a shared state. Request locking has been introduced to solve this but it's still experimental.
> - Avoiding Overloading the Source Website: When parallelizing a web scraper, it's crucial to ensure that the scraping activity does not overload the source website. Running multiple scrapers in parallel can significantly increase the number of requests to the website, which might not only affect the website's performance but also lead to your IP being blocked.

Formatted this way, I know what this is about, what it solves, and what are the cons. If I'm still interested after weighing the pros and cons, *then* I proceed to read the how-to.